examples/client/Locomotion/src/pages/ActiveRide/index.js

```diff
@@ -191,6 +198,12 @@ const RidePage = ({ mapSettings, navigation }) => {
     }
   }, [isExpanded]);
 
+  useEffect(() => {
+    if (ride.state === RIDE_STATES.CANCELED && ride.canceledBY !== user.id) {
+      setRidePopup(RIDE_POPUPS.RIDE_CANCELED_BY_DISPATCHER);
+    }
+  }, [ride.state]);
```

Comment:
don't listen to ride.state, you have the state map to function map, do it from there